tests/Integration/Core/Translation/Storage/Provider/CoreDomainCatalogueLayersProviderTest.php

```diff
@@ -0,0 +1,271 @@
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace Tests\Integration\Core\Translation\Storage\Provider;
+
+use Generator;
+use PrestaShop\PrestaShop\Core\Language\LanguageRepositoryInterface;
+use PrestaShop\PrestaShop\Core\Translation\Storage\Provider\CoreCatalogueLayersProvider;
+use PrestaShop\PrestaShop\Core\Translation\Storage\Provider\Definition\CoreDomainProviderDefinition;
+use PrestaShop\PrestaShop\Core\Translation\TranslationRepositoryInterface;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+use Symfony\Component\Translation\MessageCatalogue;
+
+/**
+ * Test the core translations provider filtering by domain
+ */
+class CoreDomainCatalogueLayersProviderTest extends KernelTestCase
+{
+    /**
+     * @var string
+     */
+    protected $translationsDir;
+
+    protected function setUp()
+    {
+        self::bootKernel();
+        $this->translationsDir = self::$kernel->getContainer()->getParameter('test_translations_dir');
+    }
+
+    /**
+     * Test it loads a XLIFF catalogue from the locale's `translations` directory
+     *
+     * @dataProvider getValuesForLoadCatalogueFromXliff
+     */
+    public function testItLoadsCatalogueFromXliffFilesInLocaleDirectory(string $domain, array $expectedCatalogue): void
+    {
+        // load catalogue from translations/fr-FR
+        $catalogue = $this->getFileTranslatedCatalogue($domain, 'fr-FR');
+
+        // verify all catalogues are loaded
+        $this->assertResultIsAsExpected($expectedCatalogue, $catalogue);
+    }
+
+    public function getValuesForLoadCatalogueFromXliff(): Generator
+    {
+        yield [
+            // domain
+            'AdminActions',
+            // expectedCatalogue
+            [
+                'AdminActions' => [
+                    'count' => 90,
+                    'translations' => [
+                        'Save and stay' => 'Enregistrer et rester',
+                        'Uninstall' => 'Désinstaller',
+                    ],
+                ],
+            ],
+        ];
+
+        yield [
+            // domain
+            'ModulesCheckpaymentAdmin',
+            // expectedCatalogue
+            [
+                'ModulesCheckpaymentAdmin' => [
+                    'count' => 15,
+                    'translations' => [
+                        'The "Payee" and "Address" fields must be configured before using this module.' => 'Les champs "Nom du bénéficiaire" et "Adresse" doivent être configurés avant d\'utiliser ce module.',
+                        'No currency has been set for this module.' => 'Aucune devise disponible pour ce module',
+                    ],
+                ],
+            ],
+        ];
```

Comment:
Interesting way of doing this :)

FYI you can also do it like this:

```suggestion
        return [
        	// core domain 
         	[
            	'domain' => 'AdminActions',
				'expectedCatalogue' => [
                	'AdminActions' => [
	                	'count' => 90,
	                    'translations' => [
	                        'Save and stay' => 'Enregistrer et rester',
	                        'Uninstall' => 'Désinstaller',
	                	],
	            	],
	        	],
			],
			// module domain
        	[
	            'domain' => 'ModulesCheckpaymentAdmin',
	            'expectedCatalogue' => [
					'ModulesCheckpaymentAdmin' => [
				    	'count' => 15,
				        'translations' => [
				        	'The "Payee" and "Address" fields must be configured before using this module.' => 'Les champs "Nom du bénéficiaire" et "Adresse" doivent être configurés avant d\'utiliser ce module.',
				            'No currency has been set for this module.' => 'Aucune devise disponible pour ce module',
			          	],
		          	],
		        ],
            ],
        ];
```